[WPE] WPE Platform: use gbm_import_fd_modifier_data in WPEBufferDMABuf.import_to_pixels

lauromoura opened the pull request:
#### ec2531af5f5f7988829cb9c7a7490e7c8171eb0f
<pre>
[WPE] WPE Platform: use gbm_import_fd_modifier_data in WPEBufferDMABuf.import_to_pixels
<a href="https://bugs.webkit.org/show_bug.cgi?id=299575">https://bugs.webkit.org/show_bug.cgi?id=299575</a>

Reviewed by Carlos Garcia Campos.

GBM offers the struct gbm_import_fd_modifier_data that we can use with
gbm_bo_import to let it know more details about the format of the
buffer, like modifier flags and multiple planes. Using it allows to
read pixels from non-linear (e.g. tiled ) or multi-planar buffers,
leaving only the pixel format as the only remaining constraint.

Also, we don&apos;t need to pass the GBM_BO_USE_RENDERING and
GBM_BO_USE_LINEAR flags to gbm_bo_import, as they&apos;re relevant only when
creating BO instances. In fact, gbm_bo_map already &quot;exposes a linear
view of the buffer even if the buffer has a non-linear modifier&quot;.

* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp:
(wpeBufferDMABufImportToPixels):

Canonical link: <a href="https://commits.webkit.org/300687@main">https://commits.webkit.org/300687@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3ea122312abb0b52ab8762e12823ba4ddeeee2a8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123072 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42786 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33483 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129728 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75187 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43510 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51381 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93551 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62085 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126023 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34678 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110145 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74181 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33651 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73245 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104388 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28527 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132459 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50022 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38079 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102051 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50398 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106366 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101907 "Found 3 new API test failures: TestWTF:CompletionHandlerDeathTest.MainThreadHandlerOnThreadAssertsDeathTest, WebKitGTK/TestInspector:/webkit/WebKitWebInspector/manual-attach-detach, TestWTF:CompletionHandlerDeathTest.ConstructionThreadHandlerOnThreadAssertsDeathTest (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25984 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47275 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25477 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/46795 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49877 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55638 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49345 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52697 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51026 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (commented):
Glad this works! I think you need to keep the old code in case there are no modifiers used (indicated by the DRM_...INVALID flag)

lauromoura committed:
[WPE] WPE Platform: use gbm_import_fd_modifier_data in WPEBufferDMABuf.import_to_pixels
https://bugs.webkit.org/show_bug.cgi?id=299575

Reviewed by Carlos Garcia Campos.

GBM offers the struct gbm_import_fd_modifier_data that we can use with
gbm_bo_import to let it know more details about the format of the
buffer, like modifier flags and multiple planes. Using it allows to
read pixels from non-linear (e.g. tiled ) or multi-planar buffers,
leaving only the pixel format as the only remaining constraint.

Also, we don't need to pass the GBM_BO_USE_RENDERING and
GBM_BO_USE_LINEAR flags to gbm_bo_import, as they're relevant only when
creating BO instances. In fact, gbm_bo_map already "exposes a linear
view of the buffer even if the buffer has a non-linear modifier".

* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp:
(wpeBufferDMABufImportToPixels):

Canonical link: https://commits.webkit.org/300687@main
